src/main/java/org/folio/fqm/migration/AbstractSimpleMigrationStrategy.java

```diff
@@ -0,0 +1,72 @@
+package org.folio.fqm.migration;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.UncheckedIOException;
+import java.util.Map;
+import java.util.UUID;
+import lombok.extern.log4j.Log4j2;
+import org.folio.fql.model.Fql;
+import org.folio.fql.service.FqlService;
+
+@Log4j2
+public abstract class AbstractSimpleMigrationStrategy implements MigrationStrategy {
+
+  /** The version migrating FROM */
+  public abstract String getSourceVersion();
+
+  /** The version migrating TO */
+  public abstract String getTargetVersion();
+
+  /** The entity types that got a new UUID */
+  protected abstract Map<UUID, UUID> getEntityTypeChanges();
+
+  /**
+   * The fields that were renamed. Keys should use their OLD entity type ID, if applicable.
+   *
+   * The special key "*" on the field map can be used to apply a template to all fields. This value
+   * should be a string with a %s placeholder, which will be filled in with the original field name.
+   */
+  protected abstract Map<UUID, Map<String, String>> getFieldChanges();
+
+  @Override
+  public boolean applies(FqlService fqlService, MigratableQueryInformation src) {
+    String srcVersion = "0";
+    if (src.fqlQuery() != null) {
+      Fql query = fqlService.getFql(src.fqlQuery());
+      if (query._version() != null) {
+        srcVersion = query._version();
+      }
+    }
+
+    return this.getSourceVersion().equals(srcVersion);
```

Comment:
Is this logic backward? I'd think the strategy would apply if the versions _don't_ match

Reply:
It's correct; the simple ones have a source A and target B; we only want to run the migration to go from A -> B. If it starts at anything other than A, other migration(s) should be done to get to A before the A->B change is made